src/main/java/org/zalando/nakadi/repository/db/EventTypeCache.java

```diff
@@ -81,71 +121,69 @@ public void removed(final String name) throws Exception {
         zkClient.delete().forPath(path);
     }
 
-    private void initParentCacheZNode(final CuratorFramework zkClient) throws Exception {
+    private Optional<CachedValue> getCached(final String name)
+            throws NoSuchEventTypeException, InternalNakadiException {
         try {
-            zkClient
-                    .create()
-                    .creatingParentsIfNeeded()
-                    .withMode(CreateMode.PERSISTENT)
-                    .forPath(ZKNODE_PATH);
-        } catch (KeeperException.NodeExistsException expected) {
-            // silently do nothing since it means that the node is already there
+            return Optional.ofNullable(eventTypeCache.get(name));
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof NoSuchEventTypeException) {
+                throw (NoSuchEventTypeException) e.getCause();
+            } else {
+                throw new InternalNakadiException("Problem loading event type", e);
+            }
         }
     }
 
-    private void addCacheChangeListener(final LoadingCache<String, EventType> eventTypeCache,
-                                        final LoadingCache<String, EventTypeValidator> validatorCache,
-                                        final PathChildrenCache cacheSync) {
-        final PathChildrenCacheListener listener = new PathChildrenCacheListener() {
-            @Override
-            public void childEvent(final CuratorFramework client, final PathChildrenCacheEvent event) throws Exception {
-                if (event.getType() == PathChildrenCacheEvent.Type.CHILD_UPDATED) {
-                    invalidateCacheKey(event);
-                } else if (event.getType() == PathChildrenCacheEvent.Type.CHILD_REMOVED) {
-                    invalidateCacheKey(event);
-                }
-            }
-
-            private void invalidateCacheKey(final PathChildrenCacheEvent event) {
-                final String path[] = event.getData().getPath().split("/");
-                final String key = path[path.length - 1];
-
-                validatorCache.invalidate(key);
-                eventTypeCache.invalidate(key);
-            }
-        };
+    public EventType getEventType(final String name) throws NoSuchEventTypeException, InternalNakadiException {
+        return getCached(name).map(CachedValue::getEventType).orElse(null);
+    }
 
-        cacheSync.getListenable().addListener(listener);
+    public EventTypeValidator getValidator(final String name) throws InternalNakadiException, NoSuchEventTypeException {
```

Comment:
If it can return null - the method should be declared as `@Nullable`, and it should be properly handled in the places where it is called. The other possibility here will be to use `orElseThrow(...)` instead.